src/cpp/shared_core/Logger.cpp

```diff
@@ -580,6 +580,11 @@ bool hasStderrLogDestination()
    return hasLogDestination<StderrLogDestination>();
 }
 
+bool needsDebugLog()
```

Comment:
Nit: This file should have the functions implemented in the same order they appear in the header file. Would you mind reordering these to that effect?